### PR TITLE
CTCP-776: Adding GET single message endpoint

### PR DESCRIPTION
**Pull request by drnaylor-oc**

### CODE REVIEW

**Comment by drnaylor-oc** [comment]
image

File: src/main/java/uk/gov/hmrc/components/GetSingleDepartureMessageForm.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            log.error("Unable to get a message: server returned unauthorised");
+            error("Unable to get a message: server returned unauthorised");
+        } catch (NotFoundException e) {
+            log.error("Unable to get a message: not found or archived");
+            error("Unable to get a message: not found or archived");
+        } catch (RequestException e){
+            log.error("Unable to submit the declaration: "+e.getMessage(), e);
+            error("Unable to get a message: an unexpected error occurred");
+        }
+    }
```
Comment (derekspalmer) [inline src/main/java/uk/gov/hmrc/components/GetSingleDepartureMessageForm.java:54]: Same here. Possibly add a finally.

**Review by derekspalmer — APPROVED** [review]
Looks fine in terms of the code, but tbh not particularly familiar with how it all hangs together!

File: src/main/java/uk/gov/hmrc/components/DepartureDeclarationForm.java [hunk 2]
```diff
@@ -40,9 +40,10 @@ protected void onSubmit() {
...
-//            info("Movement id: "+response.getId());
-//            info("Movement reference: "+response.getLinks().getSelf().getHref());
-        } catch(MessageSubmissionException e){
+            info("Movement ID: " + response.getLinks().getSelf().getHref());
+        } catch(UnauthorizedException e) {
+            log.error("Unable to submit the declaration: server returned unauthorised");
+        } catch(RequestException e) {
             log.error("Unable to submit the declaration: "+e.getMessage(), e);
             ValidationErrors errors = formatError(e);
             for ( ValidationError error : errors.getValidationErrors()) {
```
[the remark this answers is not in the document]
Comment (drnaylor-oc) [reply src/main/java/uk/gov/hmrc/components/DepartureDeclarationForm.java:49]: What would we use a finally block for?

**Pull request merged**